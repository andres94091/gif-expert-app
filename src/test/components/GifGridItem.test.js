import '@testing-library/jest-dom';
import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en <GifGridItem/>', () => {

    const title = 'este es un titulo'
    const url = 'http://localhost/algo.jpg'

    const wrapper = shallow(<GifGridItem
        title={title}
        url={url}
    />)

    test('debe renderizar el componente de manera correcta', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('debe de tener un parrafo con titulo', () => {

        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title)

    })

    test('debe tener la imagen igual al url y alt de los props', () => {

        const img = wrapper.find('img')
        expect(img.prop('src')).toBe(url)
        expect(img.prop('alt')).toBe(title)
    })

    test('debe tener animate_fadeIn', () => {
        
        const div = wrapper.find('div')
        expect(div.props().className.includes('animate__fadeIn')).toBe(true);
    })
    

})
