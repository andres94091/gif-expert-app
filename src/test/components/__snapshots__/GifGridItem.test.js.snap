// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem/> debe renderizar el componente de manera correcta 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="este es un titulo"
    src="http://localhost/algo.jpg"
  />
  <p>
    este es un titulo
  </p>
</div>
`;
